src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_NullablePublicOnly.cs

```diff
@@ -0,0 +1,199 @@
+        }
+
+        [Fact]
+        public void EmptyProject()
+        {
+            var source = @"";
+            var options = WithNonNullTypesTrue().WithMetadataImportOptions(MetadataImportOptions.All);
+            var parseOptions = TestOptions.Regular8;
+
+            var comp = CreateCompilation(source, options: options, parseOptions: parseOptions);
+            CompileAndVerify(comp, symbolValidator: AssertNoNullablePublicOnlyAttribute);
+
+            comp = CreateCompilation(source, options: options, parseOptions: parseOptions.WithFeature("nullablePublicOnly"));
+            CompileAndVerify(comp, symbolValidator: AssertNullablePublicOnlyAttribute);
+        }
+
+        [Fact]
+        public void CSharp7_NoAttribute()
+        {
+            var source =
+@"public class A<T>
+{
+}
+public class B : A<object>
+{
+}";
+            var options = TestOptions.ReleaseDll.WithMetadataImportOptions(MetadataImportOptions.All);
+            var parseOptions = TestOptions.Regular7;
+
+            var comp = CreateCompilation(source, options: options, parseOptions: parseOptions);
+            CompileAndVerify(comp, symbolValidator: AssertNoNullablePublicOnlyAttribute);
+
+            comp = CreateCompilation(source, options: options, parseOptions: parseOptions.WithFeature("nullablePublicOnly"));
+            CompileAndVerify(comp, symbolValidator: AssertNoNullablePublicOnlyAttribute);
+        }
+
+        [Fact]
+        public void NullableEnabled()
+        {
+            var source =
+@"public class A<T>
+{
+}
+public class B : A<object?>
+{
+}";
+            var options = WithNonNullTypesTrue().WithMetadataImportOptions(MetadataImportOptions.All);
+            var parseOptions = TestOptions.Regular8;
+
+            var comp = CreateCompilation(source, options: options, parseOptions: parseOptions);
+            CompileAndVerify(comp, symbolValidator: AssertNoNullablePublicOnlyAttribute);
+
+            comp = CreateCompilation(source, options: options, parseOptions: parseOptions.WithFeature("nullablePublicOnly"));
+            CompileAndVerify(comp, symbolValidator: AssertNullablePublicOnlyAttribute);
+        }
+
+        [Fact]
+        public void NullableEnabled_NoPublicMembers()
+        {
+            var source =
+@"class A<T>
+{
+}
+class B : A<object?>
+{
+}";
+            var options = WithNonNullTypesTrue().WithMetadataImportOptions(MetadataImportOptions.All);
+            var parseOptions = TestOptions.Regular8;
+
+            var comp = CreateCompilation(source, options: options, parseOptions: parseOptions);
+            CompileAndVerify(comp, symbolValidator: AssertNoNullablePublicOnlyAttribute);
+
+            comp = CreateCompilation(source, options: options, parseOptions: parseOptions.WithFeature("nullablePublicOnly"));
+            CompileAndVerify(comp, symbolValidator: AssertNullablePublicOnlyAttribute);
```

Comment:
>symbolValidator: AssertNullablePublicOnlyAttribute); [](start = 35, length = 52)

consider validating that no NullablePublicOnly attribute appears on source symbols (`validator: AssertNoNullablePublicOnlyAttribute`)